core/svelte-compiler.js

```diff
@@ -74,8 +75,8 @@ SvelteCompiler = class extends CachingCompiler {
     try {
       const compiled = svelte.compile(raw, {
         filename: path,
-        name: file.getBasename()
-          .slice(0, -(extension.length + 1)) // Remove extension
+        name: basename
+          .slice(0, basename.indexOf('.')) // Remove extension
```

Comment:
Exactly. With `lastIndexOf` (or the previous implementation), `Foo.bar.html` would become `Foo.bar` which would then be sanitized to `Foo_bar` in the line below. However, this part only defines the name of the component constructor so that it appears as `Foo` in generated code and stack traces.